### DISCUSSION TITLE
Multiple Dynamic Role Assignment for new registered user

### DISCUSSION THREAD

**Discussion by usamaB** [apache/superset#18321] [Q&A / Help]
Hi, I'm not fully sure if this is already available.
I am thinking of something like setting an array of roles. Atm only one can be setup. 
e.g.
AUTH_USER_REGISTRATION_ROLE= ['role1', 'role2']

I have the following case
// This only sets SomeCustomRole and not Gamma in addition to it if the user is from domain.
AUTH_USER_REGISTRATION_ROLE_JMESPATH="(contains(['list of users'], email) && 'Admin') || (ends_with(email, '@domain.com') && 'Gamma'&& 'SomeCustomRole') || 'Gamma'"

// This gives Exception as the in the logs **LINE 3: WHERE ab_role.name = ARRAY['Gamma','SomeCustomRole']**
AUTH_USER_REGISTRATION_ROLE_JMESPATH="(contains(['list of users'], email) && 'Admin') || (ends_with(email, '@domain.com') && ['Gamma','SomeCustomRole']) || 'Gamma'"

OR if there is a  way to Extend Gamma role and add permissions to it,  then one role assignment could work. Adding all the permissions(which  Gamma role has) would be very for each new custom role.

I'm using helm chart. 
Is there anything like this available?
Thanks

**Comment by akhileshmalavalli** [comment]
Did you get this resolved @usamaB ?
